graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -460,10 +518,20 @@ abstract class BaseDataTypeGenerator(
                 .addMethod(buildMethod)
 
         javaType.build().fieldSpecs.map {
-            MethodSpec.methodBuilder(it.name)
+            val method = MethodSpec.methodBuilder(it.name)
                 .addJavadoc(it.javadoc)
                 .returns(builderClassName)
                 .addStatement("this.${it.name} = ${it.name}")
+
+            val fieldName = it.name
+            val field = fields.find { it.name.contains(fieldName) }
+
+            if (field?.isNullable == true && field.initialValue == null) {
+                method
+                    .addStatement("this.${generateBooleanFieldName(it.name)} = true")
```

Comment:
Another instance where we can use the JavaPoet formatting, e.g., `this.\$N = true`

Reply:
Updated 👍